### ISSUE TITLE
Material text interactive selection uses buttonTheme

### ISSUE THREAD

**Issue by miguelpruivo** [flutter/flutter#37550]
The Material interactive selection shouldn't use the `buttonTheme` or, at least, the theme should provide a flag to either use it in the interactive selection or not.

As it is, it's impossible to either use a custom `buttonTheme` (such setting `minWidth`, `height` and so on) without directly affecting the interactive selection that should stick with the native look & feel.

Plus, a good alternative might be having a `interactiveSelectionTheme` that would allow customize only it.

```
 Flutter (Channel unknown, v1.8.0, on Mac OS X 10.14.6 18G87, locale pt-PT)
    • Flutter version 1.8.0 at /Users/miguelruivo/DevTools/flutter
    • Framework revision 2fefa8c731 (5 weeks ago), 2019-07-01 11:33:22 -0700
    • Engine revision 45b66b722e
    • Dart version 2.4.0
```

**Comment by HansMuller** [comment]
What material library component[s] are you referring to - "interactive selection"?

**Comment by miguelpruivo** [comment]
@HansMuller I don’t know which is the correct name for the text selection options that appear (copy, cut, paste), so I thought it could be explicit enough, sorry. 😅

Anyway, if you customize your `buttonTheme` and set, let’s say, a `minWidth` and with a circular shape, because you want your app `RaisedButton`s to match that style, you’ll also directly affect the _cut, copy, paste_ option buttons from the text selection.

**Comment by HansMuller** [comment]
OK, I think I understand what you're after.  The material TextField builds a [TextSelectionControls instance](https://api.flutter.dev/flutter/material/materialTextSelectionControls.html) that ends up in the app's overlay. So it only really inherits from the app's theme.  That means a definition of ButtonTheme in the TextField's context will not apply to the text field menu. And it means that a global ButtonTheme, one that's defined as part of the overall Theme, will apply to text select controls too; even if that's not what you want.

There are a couple of ways around this: 
- We could automatically wrap the text selection controls in copies of the "in context" themes.
- We could provide a ButtonThemeData valued ThemeData property for text selection controls.
- We could make it possible to build you own text selection controls and/or wrap the default ones.

**Comment by miguelpruivo** [comment]
@HansMuller what if you don't have any "in context" theme? Will it use the global theme?

For simplicity sake and also, to make it more customizable, I personally would like to see your second point: a property -- which can be inside the `ButtonTheme` -- that could allow styling the `TextSelectionControls` which, as of now, seems pretty unclear how to achieve so.

**Comment by miguelpruivo** [comment]
@HansMuller what if you don't have any "in context" theme? Will it use the global theme?

For simplicity sake and also, to make it more customizable, I personally would like to see your second point: a property -- which can be inside the ButtonTheme -- that could allow styling the TextSelectionControls which, as of now, seems pretty unclear on how to achieve so.

**Comment by renanyoy** [comment]
also it's only for android, iOS use the system ones

**Comment by miguelpruivo** [comment]
@renanyoy I'm afraid not. They both share theme widgets, if you set the text font scale on iOS device settings to max it will also overflow the interactive selection box.

**Comment by renanyoy** [comment]
no. for me I have white on black bubble, really system ones on iOS, nothing like the ones on android..
I have setup my button with stadium border as default theme.. and my main colors are blue or orange depending of the user rights, no white or black in the theme. and I have stadium cut/paste/select buttons on android.

**Comment by PsyOhm23** [comment]
The scale issue on iOS is already fixed here #54140 
For the current issue, text select controls doesn't uses the `ButtonThemeData` for iOS unlike Android.
I think adding a custom theme for text selection would be nice.
I have a ButtonThemeData with a `minWidth` setted to 300 which make text select controls too big.
